sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -50,23 +51,30 @@ class ReflectiveActionRouter[A <: Action](
     val inputTypeUrl = message.payload().asInstanceOf[ScalaPbAny].typeUrl
     val methodInvoker = commandHandler.lookupInvoker(inputTypeUrl)
 
-    methodInvoker match {
-      case Some(invoker) =>
-        inputTypeUrl match {
-          case ProtobufEmptyTypeUrl =>
-            invoker
-              .invoke(action)
-              .asInstanceOf[Action.Effect[_]]
-          case _ =>
-            invoker
-              .invoke(action, invocationContext)
-              .asInstanceOf[Action.Effect[_]]
-        }
-      case None if ignoreUnknown => ActionEffectImpl.Builder.ignore()
-      case None =>
-        throw new NoSuchElementException(
-          s"Couldn't find any method with input type [$inputTypeUrl] in Action [$action].")
-
+    // lookup ComponentClient
+    val maybeComponentClient = Reflect.lookupComponentClientField(action.getClass, action)
```

Comment:
There is no test for this, but @efgpinto's [PR](https://github.com/lightbend/kalix-jvm-sdk/pull/2026/files) will add a sample that will then cover it.